### PR TITLE
feat(tabstops-auto-target-page-vis): Add automatically failed tab stops target page visualization elements

### PR DESCRIPTION
**Pull request by sfoslund**
#### Details

Add visualization elements that will be leveraged for automated tab stops failure visualization in target page. Note that screenshots below were generated during tested and do not show up in any real visualizations. 

Missing element visualization: dotted red circle with red '!' in right upper corner. Once logic is added to differentiate between types of tabstops visualization elements, the tab order number will be replaced with a red 'X'. 
MissingElementsFailureVis
(screenshot of missing element visualization described above)

Failed element visualization: solid red circle with red '!' in right upper corner. This visualization will represent any automatically detected tab stop failure besides a missing element. Once logic is added to differentiate between types of tabstops visualization elements, the tab order text will be red. 
FailedElementVis
(screenshot of failed element visualization described above)

##### Motivation

Feature work.

##### Context

This PR only adds the elements themselves, they are not yet leveraged in the target page visualization. A follow up PR will add the logic to determine which failure elements need to be displayed. 

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [ not yet included in visualization ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/injected/visualization/svg-shape-factory.ts [hunk 1]
```diff
@@ -97,11 +98,42 @@ export class SVGShapeFactory {
 
         if (tabOrder != null) {
             text.innerHTML = tabOrder.toString();
+        } else {
+            text.innerHTML = 'X';
```
Comment (lisli1) [inline src/injected/visualization/svg-shape-factory.ts:102]: There is a possibility that the circle won't have any text (user toggles visualization off and then on again so the tab order disappears; errored circle would have no number in it).

File: src/injected/visualization/svg-shape-factory.ts [hunk 2]
```diff
@@ -97,11 +98,42 @@ export class SVGShapeFactory {
...
         return text;
     }
 
+    public createFailureLabel(center: Point, failureBoxConfig: FailureBoxConfig): Element {
+        const myDocument = this.drawerUtils.getDocumentElement();
+
+        const box = myDocument.createElementNS(SVGNamespaceUrl, 'rect');
+        box.classList.add('insights-highlight-text');
+        box.classList.add('failure-label');
+        box.setAttributeNS(null, 'x', (center.x + 10).toString());
```
Comment (lisli1) [inline src/injected/visualization/svg-shape-factory.ts:114]: Curious how you came up with the calculations for the position?

File: src/injected/visualization/tab-stops-formatter.ts [hunk 3]
```diff
@@ -58,6 +73,13 @@ export class TabStopsFormatter implements Formatter {
                 strokeWidth: '3',
                 strokeDasharray: '7 2',
             },
+            failureBoxConfig: {
```
Comment (lisli1) [inline src/injected/visualization/tab-stops-formatter.ts:76]: nit: the corners in the mock look slightly rounded. Could we add that here?

File: src/injected/visualization/svg-shape-factory.ts (hunk 2 above)
Comment (sfoslund) [reply src/injected/visualization/svg-shape-factory.ts:114]: Yeah, the circles are 10px in radius, so to get to the upper right corner we need to do center.x + 10 and center.y - 10. From there, these numbers have to be adjusted to account for the size of the red box, and then to center the '!' in the box.

File: src/injected/visualization/tab-stops-formatter.ts (hunk 3 above)
Comment (sfoslund) [reply src/injected/visualization/tab-stops-formatter.ts:76]: Yup! Added that so the failure labels now look like this:
RoundedCorner
(screenshot of updated failure label, red box with '!' in top right corner of tab stops circle, now with rounded corners)

**Review by lisli1 — APPROVED** [review]
LGTM

**Pull request merged**